controllers/history.controller.js

```diff
@@ -0,0 +1,67 @@
+const { History } = require('../models/history.model');
+
+const getAllHistory = async (req, res) => {
+  try {
+    const { userID } = req.user;
+    const allHistory = await History.findById(userID).populate('videos');
+    res.json({ success: true, allHistory });
+  } catch (error) {
+    res.json({ success: false, message: 'Error retrieving liked videos!', errorMessage: error.message });
+  }
+};
+
+const deleteAllHistory = async (req, res) => {
+  try {
+    const { userID } = req.user;
+    const history = await History.findById(userID);
+    await history.updateOne({ $set: { videos: [] } });
+    await history.save();
+    res.json({ success: true, history });
+  } catch (error) {
+    res.json({ success: false, message: 'Error retrieving history!', errorMessage: error.message });
+  }
+};
+
+const addToHistory = async (req, res) => {
+  try {
+    const { videoID } = req.params;
+    const { userID } = req.user;
+    const history = await History.findById(userID);
+
+    if (!history) {
+      const newHistory = new History({
+        _id: userID,
+        videos: [{ _id: videoID }],
```

Comment:
hey pranshu, you created an emoty history as soon as user is signed up, so line number 29 will surely give a result.
I think checking whether history exists or not for the user is redundant here